File: server/src/main/java/org/dan/jadalnia/app/user/Uid.java
package org.dan.jadalnia.app.user;

import com.fasterxml.jackson.annotation.JsonCreator;
import org.dan.jadalnia.sys.type.number.ImmutableNumber;

public class Uid extends ImmutableNumber {
    @JsonCreator
    public Uid(int id) {
        super(id);
    }

    // jax-rsp
    @JsonCreator
    public static Uid valueOf(String s) {
        return new Uid(Integer.valueOf(s));
    }

    public static Uid of(int id) {
        return new Uid(id);
    }
}

File: kot/src/main/kotlin/org/dan/jadalnia/app/user/Uid.kt
package org.dan.jadalnia.app.user;

import com.fasterxml.jackson.annotation.JsonCreator;
import org.dan.jadalnia.sys.type.number.ImmutableNumber;

class Uid @JsonCreator constructor(id: Int): ImmutableNumber(id) {
    companion object {
        // jax-rsp
        @JvmStatic @JsonCreator
        fun valueOf(s: String) = Uid(Integer.valueOf(s))

        @JvmStatic
        fun of(id: Int) = Uid(id)
    }
}
